### ISSUE TITLE
Multiple instances - livereload doesn't work

### ISSUE THREAD

**Issue by sdousley** [danlynn/ember-cli#20]
Hi

I have 3 ember applications, and when I configure this image using docker-compose, if I have 1 server running, then this works perfectly fine.  

If I add a second in, one of them works fine, but on the other, livereload stops working, and in fact, refreshing the page doesn't even load the changes.  

The only way I can load the changes is to restart docker with "docker-compose stop && docker-compose up".

I have them all configured to use different ports on my local machine.  

Here's the 3 containers cut from the docker-compose.yml

https://gist.github.com/sdousley/a865d0074648b711e074afb188a7af47

**Comment by danlynn** [comment]
You probably need to configure each of the ember instances to run of separate ports themselves instead of having docker remap the ports on the fly.  This is because the client app running in the browser doesn’t know the correct ports to be talking back with its own container.  If you treat this like running 3 instances on the same machine where you have to config the ports at the ember level then everything should work fine.

**Comment by sdousley** [comment]
Hi

I'm not 100% certain this is the issue.  The reason being that with all 3 configured in docker-compose, I can access all 3 ember instances.

I can also access all 3 with livereload when they're configured individually, just can't get live reload to work (reliably) when more than 1 are configured.

It's a really odd circumstance, and I don't appear to be getting any (obvious) errors in bringing the containers up, or in the console.

**Comment by danlynn** [comment]
I think the live reload is confused because the app is internally trying to establish a connection to the port defined be default within ember-cli.

You can set that port for each instance with

{
  "live-reload-port": 35729
}

In the .ember-cli config:

https://ember-cli.com/#runtime-configuration <https://ember-cli.com/#runtime-configuration>

**Comment by sdousley** [comment]
Hi yeah, I did that and set it to the ports as defined in the Dockerfile (each being different).

It's really odd though because it's not so much live reload that doesn't work, but reloading full stop.

If I force-reload a page the changes aren't picked up until I restart docker, which makes no sense to me.

I am not sure what the 3rd port defined in the Dockerfile is.

Is there any way for me to see the ember output on the containers?

**Comment by sdousley** [comment]
Hi

I had that all setup, and it still wouldn't work.

However, upon investigating docker logs for the container that doesn't work, I've found the issue with this, and it's not actually anything to do with this docker, it's a setting on my host machine that I needed to change to allow this to work.

It's to do with inotify limits, in my case, the max_user_watches setting to be more specific.

On my laptop, this was set to 8192, I googled how to increase it, and found something that showed how to increase it to 524288, so I did that, and all is good now.

The command that I ran in particular was:

`sudo echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p`
